pkg/health/condition/check_backup_ready.go

```diff
@@ -37,101 +38,194 @@ const (
 	BackupFailed string = "BackupFailed"
 	// Unknown is a constant that means that the etcd backup status is currently not known
 	Unknown string = "Unknown"
-	// ConditionNotChecked is a constant that means that the etcd backup status has not been updated or rechecked
-	ConditionNotChecked string = "ConditionNotChecked"
+	// NotChecked is a constant that means that the etcd backup status has not been updated or rechecked
+	NotChecked string = "ConditionNotChecked"
 )
 
 func (a *backupReadyCheck) Check(ctx context.Context, etcd druidv1alpha1.Etcd) Result {
-	//Default case
-	result := &result{
-		conType: druidv1alpha1.ConditionTypeBackupReady,
-		status:  druidv1alpha1.ConditionUnknown,
-		reason:  Unknown,
-		message: "Cannot determine etcd backup status",
-	}
-
 	// Special case of etcd not being configured to take snapshots
 	// Do not add the BackupReady condition if backup is not configured
 	if etcd.Spec.Backup.Store == nil || etcd.Spec.Backup.Store.Provider == nil || len(*etcd.Spec.Backup.Store.Provider) == 0 {
 		return nil
 	}
 
-	//Fetch snapshot leases
-	var (
-		fullSnapErr, incrSnapErr error
-		fullSnapLease            = &coordinationv1.Lease{}
-		deltaSnapLease           = &coordinationv1.Lease{}
+	// Fetch snapshot leases
+	fullSnapshotLease, err := a.fetchLease(ctx, etcd.GetFullSnapshotLeaseName(), etcd.Namespace)
+	if err != nil {
+		return createBackupConditionResult(
+			druidv1alpha1.ConditionUnknown, Unknown,
+			fmt.Sprintf("Unable to fetch full snap lease. %s", err.Error()),
+		)
+	}
+
+	deltaSnapSnapshotLease, err := a.fetchLease(ctx, etcd.GetDeltaSnapshotLeaseName(), etcd.Namespace)
+	if err != nil {
+		return createBackupConditionResult(
+			druidv1alpha1.ConditionUnknown, Unknown,
+			fmt.Sprintf("Unable to fetch delta snap lease. %s", err.Error()),
```

Comment:
>do you wish to include err.Error() as part of the message or just log it using logger.Error? How large is err.Error()

It's safer to add the error to the condition, just so that it's visible to an operator without having to sift through logs. Even gardener shoot conditions for instance store the error message in the condition, which are printed to the dashboard as well. It's quite helpful for operators and users alike.

Reply:
> If you see the reason clearly indicates what that condition is for. So having a reason as Unknown would be un-qualified and therefore very hard to reason or disambiguate or even process later. In your proposal one has to look at the message to learn more about the condition which is a departure from the existing set of conditions.

I'll try to add more meaningful reason strings then.

Reply:
>Check method returns a single Result. If there are problems fetching both the leases then you will always return the condition with message for delta, thereby masking full-snapshot lease condition message. Would it make sense to have different conditions - one for delta and another for full snapshot?

I've handled such cases specifically so that the full snapshot lease error does not get masked by delta snapshot lease error. If both leases have errors, both are captured in the condition, such as [this](https://github.com/shreyas-s-rao/etcd-druid/blob/44cc1ae3558cabff57298039666ba5423dc444a3/pkg/health/condition/check_backup_ready.go#L159) and [this](https://github.com/shreyas-s-rao/etcd-druid/blob/44cc1ae3558cabff57298039666ba5423dc444a3/pkg/health/condition/check_backup_ready.go#L221). 
Looks like only the case of failing to fetch the leases needs to be handled more robustly. I'll handle this then, thanks

Reply:
>What happens when lease itself is NotFound? Should you not have a different message indicating that the lease itself is missing?

Right now, it's a blanket message of `Unable to fetch full/delta snap lease: <error-string>`, which is still technically correct. The error string holds the reason as to why the fetch failed, and it will specify that lease not found. If you want, I can separate out the lease-not-found case and use a separate `Reason` string for that like `LeaseNotFound`. WDYT?